arrow/src/record_batch.rs

```
@@ -353,6 +354,35 @@ impl RecordBatch {
         let schema = Arc::new(Schema::new(fields));
         RecordBatch::try_new(schema, columns)
     }
+
+    /// Concatenates `batches` together into a single record batch.
+    pub fn concat(schema: &SchemaRef, batches: &[Self]) -> Result<Self> {
+        if batches.is_empty() {
+            return Ok(RecordBatch::new_empty(schema.clone()));
+        }
+        let field_num = schema.fields().len();
+        if let Some((i, _)) = batches
+            .iter()
+            .enumerate()
+            .find(|&(_, batch)| batch.num_columns() < field_num)
```

I could not confirm if a batch of more columns than schema could be allowed to be concatenated, as below test case [concat_record_batches_of_similar_schemas](https://github.com/apache/arrow-rs/pull/537/files#diff-9841f61221a67c626eb60a49f2aeb7de6c1a16ab478bc7cb2675f5807b32b8e6R700).

---

I think if the batches don't have the same schema, an error should result, even if there is some plausible behavior.

My rationale is that it is likely a user error if the schemas don't match, so failing fast will help the programmer identify the problem more easily

So perhaps this function should check `batch.schema() != schema` (aka compare the entire schema for each record batch)